crates/sui-genesis-builder/src/lib.rs

```diff
@@ -1079,6 +1079,16 @@ pub fn generate_genesis_system_object(
         pt,
     )?;
 
+    // update the value of the clock to match the chain start time
+    {
```

Comment:
What exactly does this introduce to existing logic which we did not have before?

Reply:
This is fixing an oversight with how the onchain Clock object is initialized. It doesn't impact mainnet but cleans up the issue when initializing new chains that the clock and the epoch start time disagree with eachother